src/Discord.Net.Commands/Info/CommandInfo.cs

```diff
@@ -176,12 +175,22 @@ public Task<ExecuteResult> ExecuteAsync(ICommandContext context, ParseResult par
             }
         }
 
-        private async Task ExecuteAsyncInternal(ICommandContext context, object[] args, IServiceProvider services)
+        private async Task<IResult> ExecuteAsyncInternal(ICommandContext context, object[] args, IServiceProvider services)
         {
             await Module.Service._cmdLogger.DebugAsync($"Executing {GetLogText(context)}").ConfigureAwait(false);
             try
             {
-                await _action(context, args, services).ConfigureAwait(false);
+                var task = _action(context, args, services);
+                if (task is Task<IResult> returnTask)
```

Comment:
I might be crazy, but this doesn't seem like it actually works. A quick test with [this code](https://gist.github.com/Joe4evr/1ba4e5abd7d7f122244cd7d681a45529) seems to print `Nope` every time.
It's possible to do `if (task is Task<RuntimeResult> resultTask)` instead, and not need the second check.